### PR TITLE
DEL-379 Adds cop for adding comment for migration best practices documentation to migration files

### PR DESCRIPTION
**Pull request by kdurr**
## [DEL-379](https://ezcater.atlassian.net/browse/DEL-379) What did we change?
- Added a cop to enforce having a comment with a link to the ezCater migration best practices doc.
- Tagged github's suggested peeps. Ignore if you have no opinions!
- Note: I have no idea if I did this correctly...

## Why are we doing this?
We had a crit-sit on the delivery squad a few months back that was caused by a mix up in migrations. We want to make sure that everyone follows the ezCater best practices for writing migrations and this will serve as a reminder with every new migration added.

## How was it tested?
- [x] Specs
- [ ] Locally

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ This gem is moving onto its own [Semantic Versioning](https://semver.org/) schem
 
 Prior to v1.0.0 this gem was versioned based on the `MAJOR`.`MINOR` version of RuboCop. The first release of the ezcater_rubocop gem was `v0.49.0`.
 
+## v2.1.0
```
Comment (kdurr) [inline CHANGELOG.md:9]: Is this the correct version?

File: lib/rubocop/cop/ezcater/migration_documentation.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      # @example
+      #   # bad
+      #   # frozen_string_literal: true
+      #
+      #   class CreateUsers < ActiveRecord::Migration[5.1]
+      #     def change
+      #
+      #   # good
+      #   # frozen_string_literal: true
+      #   # https://ezcater.atlassian.net/wiki/spaces/POL/pages/53248012/Adding+a+migration
```
Comment (zoso10) [inline lib/rubocop/cop/ezcater/migration_documentation.rb:17]: will this conflict with existing cops? i think we require a newline after the magic frozen string comment

File: lib/rubocop/cop/ezcater/migration_documentation.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        MSG = "Missing the comment `#{MIGRATION_DOC_COMMENT}` " \
+              "at the top of your migration file."
+
+        def investigate(processed_source)
+          return if processed_source.tokens.empty?
+
+          ensure_comment(processed_source)
+        end
+
+        def autocorrect(_node)
```
Comment (zoso10) [inline lib/rubocop/cop/ezcater/migration_documentation.rb:35]: :raised_hands: for autocorrect

File: lib/rubocop/cop/ezcater/migration_documentation.rb (hunk 2 above)
Comment (kdurr) [reply lib/rubocop/cop/ezcater/migration_documentation.rb:17]: ohhh good to know. well that's fun

Comment (kdurr) [reply lib/rubocop/cop/ezcater/migration_documentation.rb:17]: let me investigate and make changes if needed

**Comment by kdurr** [comment]
>would this cause failures for existing migration files?

that would be most annoying so i hope not. i'll need to try to test it out

> i wonder if this is better suited in ezcater_migrations gem as part of the development checks 
urging the person to reference the doc then

that's a great idea. i have no strong opinions but having this there makes sense to me.

**Comment by tvec** [comment]
> rubocop will ensure that folks are aware of that document, but i wonder if it's too late in the feedback loop of checks. i wonder if this is better suited in ezcater_migrations gem as part of the development checks urging the person to reference the doc then

👍 to this

 I'll often ignore comments (especially if they come from atlassian 😅) and what's nice about the migration gem is I believe it can give more targeted DB feedback to the user

File: CHANGELOG.md (hunk 1 above)
Comment (kraila) [reply CHANGELOG.md:9]: if it requires manual changes to existing migration files, I would count that as not backwards compatible, and bump major version. But, if it leaves existing migrations as is, this looks correct.

**Comment by kdurr** [comment]
Thank you all for your comments 🥇 
I'm going to close this and follow Tylers advice to have a message show up in the ezcater_migrations repo instead.

**Pull request closed without merging**